src/object/srv_ec.c

```diff
@@ -106,10 +128,32 @@ obj_ec_rw_req_split(struct obj_rw_in *orw, struct obj_ec_split_req **split_req)
 			D_ASSERT(iod->iod_type == DAOS_IOD_SINGLE);
 			idx = 0;
 			split_iod->iod_nr = 1;
+			if (with_csums) {
+				struct dcs_csum_info	*ci, *split_ci;
+
+				D_ASSERT(split_iod_csum->ic_nr == 1);
+				ci = &split_iod_csum->ic_data[0];
+				if (ci->cs_nr > 1) {
+					/* evenly distributed singv */
+					D_ASSERT(ci->cs_nr == tgt_max_idx + 1);
+					split_ci = &req->osr_singv_cis[i];
+					*split_ci = *ci;
+					split_iod_csum->ic_data = split_ci;
+					split_ci->cs_nr = 1;
+					split_ci->cs_csum +=
+						tgt_max_idx * ci->cs_len;
+					split_ci->cs_buf_len = ci->cs_len;
+				}
```

Comment:
Ideally, I hope we can remove these ec stuff on server side once we can generalize the compound RPC handling.

Reply:
seems to me we have to with some EC logic at leader server before dispatching the request.
and at target server, currently there is also a few EC logic for singv EC.